File: routeMap/app/src/main/java/ru/android/zheka/coreUI/IBaseFragment.java
package ru.android.zheka.coreUI;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

public interface IBaseFragment extends IActivity{
    void showError(Throwable throwable);

    void switchToFragment(int fragmentId, @NonNull Fragment fragment);

    FragmentManager getManager();

//    void nextView() throws InstantiationException, IllegalAccessException;
//    void net
}

File: routeMap/app/src/main/java/ru/android/zheka/coreUI/IBaseFragment.kt
package ru.android.zheka.coreUI

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

interface IBaseFragment :IActivity{
    override val activity: AppCompatActivity
    override fun showError(throwable: Throwable)
    override fun switchToFragment(fragmentId: Int, fragment: Fragment)
    override val manager: FragmentManager?
    override fun removeFragment(fragment: Fragment)
}